### PR TITLE
Refactor BaseChildImmunization class

### PR DESCRIPTION
**Pull request by ndegwamartin**
- Refactor for testability and maintainability

### CODE REVIEW

**Review by ekigamba — CHANGES_REQUESTED** [review]

File: opensrp-child/src/main/java/org/smartregister/child/contract/ChildImmunizationContract.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+        void showProgressDialog(String title, String message);
+
+        void hideProgressDialog();
+
+        void showGrowthDialogFragment(Map<String, List> growthMonitoring);
+
+        CommonPersonObjectClient getChildDetails();
+
+        String getString(int updating_dialog_title);
```
Comment (ekigamba) [inline opensrp-child/src/main/java/org/smartregister/child/contract/ChildImmunizationContract.java:33]: Fix variable naming here

**Review by ekigamba — CHANGES_REQUESTED** [review]

File: opensrp-child/src/main/java/org/smartregister/child/util/ChildDbUtils.java [hunk 2]
```diff
@@ -46,10 +44,7 @@ public static CommonPersonObjectClient fetchCommonPersonObjectClientByBaseEntity
 
         if (childDetails != null) {
 
-            String firstName = Strings.nullToEmpty(childDetails.get(Constants.KEY.FIRST_NAME));
-            String lastName = Strings.nullToEmpty(childDetails.get(Constants.KEY.LAST_NAME));
-
-            commonPersonObjectClient = new CommonPersonObjectClient(baseEntityId, childDetails, Utils.getName(firstName, lastName));
+            commonPersonObjectClient = new CommonPersonObjectClient(baseEntityId, childDetails, Constants.KEY.CHILD);
```
[the remark this answers is not in the document]
Comment (ndegwamartin) [reply opensrp-child/src/main/java/org/smartregister/child/util/ChildDbUtils.java:47]: This is meant to be the entity name e.g. mother, father e.t.c

File: opensrp-child/src/test/java/org/smartregister/child/presenter/BaseChildImmunizationPresenterTest.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+        PowerMockito.mockStatic(GrowthMonitoringLibrary.class);
+        PowerMockito.when(GrowthMonitoringLibrary.getInstance()).thenReturn(growthMonitoringLibrary);
+        Mockito.doReturn(heightRepository).when(growthMonitoringLibrary).heightRepository();
+
+        Mockito.doReturn(new ArrayList<>(Arrays.asList(new String[]{"3.9", "4.0", "4.6", "5.1"}))).when(heightRepository).findByEntityId(TEST_BASE_ENTITY_ID);
+
+        List<Height> heightList = presenter.getAllHeights(childDetails);
+        Assert.assertNotNull(heightList);
+        Assert.assertEquals(5, heightList.size());
```
[the remark this answers is not in the document]
Comment (ndegwamartin) [reply opensrp-child/src/test/java/org/smartregister/child/presenter/BaseChildImmunizationPresenterTest.java]: Done

File: opensrp-child/src/main/java/org/smartregister/child/contract/ChildImmunizationContract.java (hunk 1 above)
Comment (ndegwamartin) [reply opensrp-child/src/main/java/org/smartregister/child/contract/ChildImmunizationContract.java]: Done

**Review by ekigamba — APPROVED** [review]

**Pull request merged**